modules/gdscript/gdscript_parser.cpp

```diff
@@ -127,6 +127,7 @@ GDScriptParser::GDScriptParser() {
 	register_annotation(MethodInfo("@export_exp_easing", { Variant::STRING, "hint1" }, { Variant::STRING, "hint2" }), AnnotationInfo::VARIABLE, &GDScriptParser::export_annotations<PROPERTY_HINT_EXP_EASING, Variant::FLOAT>, 2);
 	register_annotation(MethodInfo("@export_color_no_alpha"), AnnotationInfo::VARIABLE, &GDScriptParser::export_annotations<PROPERTY_HINT_COLOR_NO_ALPHA, Variant::COLOR>);
 	register_annotation(MethodInfo("@export_node_path", { Variant::STRING, "type" }), AnnotationInfo::VARIABLE, &GDScriptParser::export_annotations<PROPERTY_HINT_NODE_PATH_VALID_TYPES, Variant::NODE_PATH>, 1, true);
+	register_annotation(MethodInfo("@export_type_string"), AnnotationInfo::VARIABLE, &GDScriptParser::export_annotations<PROPERTY_HINT_TYPE_STRING, Variant::STRING>);
```

Comment:
I'm not sure what's the use case for this export hint, or whether the name properly conveys it. I think the hint for `@export` was a bug, not an intended feature, so it may not need to be kept.

Reply:
I'm surprised that `&GDScriptParser::export_annotations<PROPERTY_HINT_TYPE_STRING, Variant::STRING>` makes it an export hint for selecting an Object type though, that's pretty weird.

Reply:
To be honest I didn't think too much about this, but I agree that I found it strange to have an object type selection. I never saw this in 3.2 but maybe it was already possible and I missed it. Maybe we should get feedback from whoever added that "feature"